recipes/network_interfaces_config.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: network_interfaces_config
+#
+# Copyright 2013-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+def network_interface_macs
+  uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs")
+  res = Net::HTTP.get_response(uri)
+  macs = res.body.gsub("/", "").split("\n")
+  macs
+end
+
+def device_name(mac)
+  cmd = Mixlib::ShellOut.new("ip -o link | grep #{mac} | awk '{print substr($2, 1, length($2) -1)}'")
+  cmd.run_command
+  cmd.stdout.gsub("\n", "")
+end
+
+def device_number(mac)
+  uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/device-number")
+  res = Net::HTTP.get_response(uri)
+  res.body
+end
+
+def device_ip(mac)
+  uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/local-ipv4s")
+  res = Net::HTTP.get_response(uri)
+  res.body
+end
+
+def gateway_address
+  cmd = Mixlib::ShellOut.new("ip r | grep default | head -n 1 | awk '{print $3}'")
+  cmd.run_command
+  cmd.stdout.gsub("\n", "")
+end
+
+def cidr_prefix_length(mac)
+  uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/subnet-ipv4-cidr-block")
+  res = Net::HTTP.get_response(uri)
+  res.body.split("/")[1]
+end
+
+def cidr_to_netmask(cidr)
+  IPAddr.new('255.255.255.255').mask(cidr).to_s
+end
+
+macs = network_interface_macs
+log "macs: #{macs}"
+
+if macs.length > 1
+
+  cookbook_file 'configure_nw_interface.sh' do
+    path '/tmp/configure_nw_interface.sh'
+    user 'root'
+    group 'root'
+    mode '0644'
+  end
+
+  cookbook_file 'apply_nw_config.sh' do
+    path '/tmp/apply_nw_config.sh'
+    user 'root'
+    group 'root'
+    mode '0644'
+  end
+
+  gw_address = gateway_address
+  log "gw_address: #{gw_address}"
+
+  # Configure nw interfaces
+  macs.each do |mac|
+    device_number = device_number(mac)
```

Comment:
minor: we can avoid the definition of these variables and call the function directly when populating the environment parameter.